src/point.jl

```diff
@@ -2,13 +2,49 @@
 "Abstract type for ASPRS LAS point data record formats 0 - 3"
 abstract type LasPoint end
 
-function Base.show(io::IO, pointdata::Vector{T}) where T <: LasPoint
+"""
+Custom PointVector struct for memory mapped LasPoints.
+Inspiration taken from UnalignedVector.jl
+and extended it with custom indexing and packing.
+"""
+struct PointVector{T} <: AbstractArray{T,1}
+    data::Vector{UInt8}
+    n::Int
+    pointsize::Int
+
+    function PointVector{T}(data::Vector{UInt8}) where {T}
+        pointsize = packed_sizeof(T)
+        n = length(data) ÷ pointsize
+        n*pointsize == length(data) || throw(DimensionMismatch("length(data) should be a multiple of $(pointsize), got $(length(data))"))
+        new{T}(data, n, pointsize)
+    end
+end
+
+Base.IndexStyle(::Type{PointVector{T}}) where {T} = IndexLinear()
+Base.length(pv::PointVector) = pv.n
+Base.size(pv::PointVector) = (length(pv),)
+
+function Base.getindex(pv::PointVector{T}, i::Int) where {T}
+    offset = (i-1) * pv.pointsize + 1
+    r = offset:offset + pv.pointsize - 1
+    io = IOBuffer(pv.data[r])
+    read(io, T)
+end
+
+function Base.setindex!(pv::PointVector{T}, val::T, i::Int) where {T}
+    offset = (i-1) * pv.pointsize + 1
+    r = offset:offset + pv.pointsize - 1
+    io = IOBuffer(pv.data[r])
+    write(io, val)
```

Comment:
Does this ever get written to the underlying array? Isn't `pv.data[r]` a copy?

Reply:
Good catch, I'll write some tests for this to see if this occurs.

Reply:
I disabled this for now, as writing to the PointVector is writing to a read only memory mapped file.